mysql-test/suite/encryption/r/filekeys_named_pipe.result

```diff
@@ -0,0 +1,31 @@
+install soname 'file_key_management';
+Warnings:
+Note	1	Read from /open-source/mariadb/build/mysql-test/var/tmp/fifo.key , filesize: 105B,max key file size :1048576B 
```

Comment:
This is a path on your specific computer so this test will always fail our CI. You will probably need `--replace-regex` in your test for this.

Reply:
Allocating 1MB is a little expensive , how about if key size < 0.5MB , then we create a  buffer which size is equal to file real size , copy data to that buffer  then free 1MB buffer

Reply:
Your suggestion that read file detecing if the file is a named pipe maybe bring some operating system compatibility issue, 
different detect strategy for different os .    
I think it is not a problem about allocating 1MB buffer  , because this buffer will free immediately after reading.